core/server/services/email-analytics/lib/event-processor.js

```diff
@@ -0,0 +1,230 @@
+const moment = require('moment');
+
+class EmailAnalyticsEventProcessor {
+    constructor({db, logging}) {
+        this.db = db;
+        this.logging = logging || console;
+
+        // avoid having to query email_batch by provider_id for every event
+        this.providerIdEmailIdMap = {};
+    }
+
+    process(event) {
```

Comment:
Think it's worth marking this method as `async` so it's clear for the caller that the result would be a Promise in most cases.